kong/dao/dao.lua

```diff
@@ -90,16 +89,13 @@ function DAO:new(db, model_mt, schema, constraints)
   self.constraints = constraints
 end
 
-function DAO:cache_key(...)
-  local t = utils.pack(...)
-
-  for i = 1, t.n do
-    if t[i] == nil then
-      t[i] = ""
-    end
-  end
-
-  return fmt("%s:%s", self.table, concat(t, ":"))
+function DAO:cache_key(arg1, arg2, arg3, arg4, arg5)
+  return fmt("%s:%s:%s:%s:%s:%s", self.table,
+             arg1 == nil and "" or arg1,
+             arg2 == nil and "" or arg2,
+             arg3 == nil and "" or arg3,
+             arg4 == nil and "" or arg4,
+             arg5 == nil and "" or arg5)
```

Comment:
I've been thinking a bit about this, and about how users can request `DELETE /cache/{cache_key}` to purge an entity from the cache. The `cache_key` is already a bit hard to figure out (we'll probably have to make a sugar endpoint that accepts an entity instead), I think the empty colons (`:::`) that can appear increase the complexity of the cache key.

Instead, this implementation should preserve the previous behavior (slightly less complicated keys) with  the same performance:
```lua
local function cache_key(arg1, arg2, arg3, arg4, arg5)
  return string.format("%s%s%s%s%s", "plugins",
                       arg1 ~= nil and ":" .. arg1 or "",
                       arg2 ~= nil and ":" .. arg2 or "",
                       arg3 ~= nil and ":" .. arg3 or "",
                       arg4 ~= nil and ":" .. arg4 or "",
                       arg5 ~= nil and ":" .. arg5 or "")
end
```

Thoughts?